qutip/tests/solver/heom/test_bofin_solvers.py

```diff
@@ -876,6 +942,107 @@ def test_discrete_level_model_fermionic_bath_with_decoupled_bosonic_bath(
         else:
             assert_raises_steady_state_time_dependent(hsolver)
 
+    @pytest.mark.parametrize(['evo'], [
+        pytest.param("qobj"),
+        pytest.param("qobjevo_const"),
+        pytest.param("qobjevo_timedep"),
+    ])
+    @pytest.mark.parametrize(['liouvillianize'], [
+        pytest.param(False, id="hamiltonian"),
+        pytest.param(True, id="liouvillian"),
+    ])
+    def test_discrete_level_model_fermionic_bath_with_coupled_bosonic_bath(
+        self, evo, liouvillianize
+    ):
+        dlm = DiscreteLevelCurrentModel(
+            gamma=0.01, W=1, T=0.5, lmax=1, e1=0.3, theta=0.5,
+        )
+        bosonic_mode = BosonicMode(
+            N=4, Omega=0.2, Lambda=0.1, gamma_b=0.1,
+        )
+
+        dlm_ref = DiscreteLevelCurrentModel(
+            gamma=0.01, W=1, T=0.5, lmax=1, e1=0.3, theta=0.5,
+            bosonic_mode=bosonic_mode,
+        )
+
+        options = {
+            "store_states": True,
+            "store_ados": True,
+            "nsteps": 15_000,
+            "rtol": 1e-7,
+            "atol": 1e-7,
+        }
+
+        # First we construct a solver with the boson modelled as part of the
+        # system and only a single Fermionic bath. This will provide the
+        # reference result for the test:
+        fermionic_bath_ref = FermionicBath(
+            dlm_ref.Q, *dlm_ref.bath_coefficients(), tag="fermionic",
+        )
+
+        hsolver_ref = HEOMSolver(
+            dlm_ref.H, [fermionic_bath_ref], 2, options=options,
+        )
+
+        # Then we construct a solver for the same system, but with the
+        # bosonic mode as a bath This is the result we would like to check:
+        H_sys = hamiltonian_to_sys(dlm.H, evo, liouvillianize)
+
+        fermionic_bath = FermionicBath(
+            dlm.Q, *dlm.bath_coefficients(), tag="fermionic",
+        )
+
+        bosonic_bath = BosonicBath(
+            dlm.Q.dag() @ dlm.Q, *bosonic_mode.bath_coefficients(),
+            combine=True, tag="bosonic",
+        )
+
+        hsolver = HEOMSolver(
+            H_sys, [bosonic_bath, fermionic_bath], 4, options=options,
+        )
+
+        # Calculate currents and occupations:
+        tlist = np.linspace(0, 1000, 300)
+        psi0 = basis(2, 0)
+
+        result_ref = hsolver_ref.run(dlm_ref.rho(psi0), tlist)
+        current_ref = [
+            dlm_ref.state_current(ado_state, tags=["fermionic"])
+            for ado_state in result_ref.ado_states
+        ]
+        sys_occupation_ref = dlm_ref.sys_occupation(
+            result_ref.states
+        )
+
+        result = hsolver.run(dlm.rho(psi0), tlist)
+        current = [
+            dlm.state_current(ado_state, tags=["fermionic"])
+            for ado_state in result.ado_states
+        ]
+        sys_occupation = dlm.sys_occupation(result.states)
+
+        np.testing.assert_allclose(current_ref, current, atol=1e-3)
```

Comment:
just a minor thing, these currents are very small, is it better to use an rtol here?  i guess the steady-state results below are checked more carefully, so this would instead just spot really large deviations in the results?

Reply:
The currents at later time are larger, but will switch back to `rtol` since the currents at early times are small. I switched to `atol` because that `rtol` can cause issues in tests with random data, but this is not such a test.